src/core.js

```diff
@@ -45,20 +54,56 @@ class Core extends Model {
 			this.theme = new InfoboxThemeData();
 		}
 
-		// store config for persistence method
-		this.persistOptions = params.persistOptions;
+		let len = routines.length;
+
+		if (len) {
```

Comment:
the constructor has gotten pretty big at this point. Let's break it out into smaller functions.